Add support for multiple field/value pairs to hset(), Fixes #705

prdx23 opened the pull request:
<!-- Thank you for your contribution! -->

## What do these changes do?

<!-- Please give a short brief about these changes. -->

This PR add in support to `hset()` to handle multiple field/value pairs, as redis 4.0.0 onwards `HSET` allows for multiple field/value pairs. In addition, `hmset()` and `hmset_dict()` are now deprecated.

I have followed the same convention as [redis-py](https://github.com/andymccurdy/redis-py/) and used the `mapping` keyword argument, I am happy to change it if needed.

redis docs reference:
`As of Redis 4.0.0, HSET is variadic and allows for multiple field/value pairs.`
 https://redis.io/commands/hset
`As per Redis 4.0.0, HMSET is considered deprecated. Please use HSET in new code.`
 https://redis.io/commands/hmset

pytest and flake8 tests passed locally.

## Are there changes in behavior for the user?

<!-- Outline any notable behaviour for the end users. -->

The api doesn't change for existing users, this only adds in additional support while keeping the previous methods working.
Also, calls to `hmset()` and `hmset_dict()` will now raise a deprecation warning.

## Related issue number

<!-- Are there any issues opened that will be resolved by merging this change? -->

Fixes #705

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is `<Name> <Surname>`.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the [`CHANGES/`](../tree/master/CHANGES) folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example:
   `Fix issue with non-ascii contents in doctest text files.`

seandstewart commented:
@prdx23 -

This looks great - can you please drop the update to `CHANGES.txt`? We're targeting a 2.0 release, since we're officially deprecating Py3.5 and may make some other breaking changes along the way. I'd like to wait to update that file until a release is ready.

prdx23 commented:
I added in the update to `CHANGES.txt` as the contributing page in the docs mentioned running `towncrier`, i have reverted it back. Also added a news fragment for the deprecation as i forgot to add it last time.

Also, thanks for reviving this library btw @seandstewart !

vishalof1mg commented:
In which version of aioredis can I find this?
